common/src/main/java/org/apache/celeborn/common/network/server/TransportChannelHandler.java

```diff
@@ -152,8 +153,18 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
 
   @Override
   public void channelRead(ChannelHandlerContext ctx, Object request) throws Exception {
-    if (request instanceof RequestMessage && !(request instanceof Heartbeat)) {
-      requestHandler.handle((RequestMessage) request);
+    if (request instanceof RequestMessage) {
+      if (request instanceof Heartbeat) {
+        logger.debug("Received heartbeat from {}.", NettyUtils.getRemoteAddress(ctx.channel()));
+        if (!enableHeartbeat) {
+          // Which disables heartbeat should send heartbeat for processing heartbeat to avoid
+          // reading idleness of which enables heartbeat.
```

Comment:
This is a bit confusing.
Perhaps something like this ?
```suggestion
          // When heartbeat is disabled, we should still response to a heartbeat if peer has 
          // heartbeat enabled - to present reading idleness
```